core/translate/optimizer.rs

```diff
@@ -260,27 +260,37 @@ fn eliminate_constants(
 
 /**
   Recursively pushes predicates down the tree, as far as possible.
+  Where a predicate is pushed determines at which loop level it will be evaluated.
```

Comment:
these added comments dont really have much to do with JOIN USING, i was just debugging in this general area and thought a few comments wouldnt hurt